Assignments
Now that we have introduced some tools for describing populations, let's try them out. First do these drills by hand, then use the Python code we've provided in the previous assignments to check your work. Keep track of your work in a Google document or markdown file that you can submit below and share with your mentor.

In [6]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

In [20]:
df = pd.DataFrame()
df['name'] = ['Greg', 'Marcia', 'Peter', 'Jan', 'Bobby', 'Cindy', 'Oliver']
df['age'] = [14, 12, 11, 10, 8, 6, 8]

Greg was 14, Marcia was 12, Peter was 11, Jan was 10, Bobby was 8, and Cindy was 6 when they started playing the Brady kids on The Brady Bunch. Cousin Oliver was 8 years old when he joined the show. What are the mean, median, and mode of the kids' ages when they first appeared on the show? What are the variance, standard deviation, and standard error?

In [22]:
#Question 1 
mean = np.mean(df['age'])
median = np.median(df['age'])

import statistics as stats
mode = stats.mode(df['age'])

var = np.var(df['age'])
std = np.std(df['age'], ddof=1)
se = std / np.sqrt(len(df['age']))

print(mean, median, mode, var, std, se)

9.857142857142858 10.0 8 6.408163265306122 2.734262327610589 1.0334540197243192


Using these estimates, if you had to choose only one estimate of central tendency and one estimate of variance to describe the data, which would you pick and why?

I would pick average age and standard deviation to describe this sample. These two statistics would provide a good picture of actors' age. 

Next, Cindy has a birthday. Update your estimates- what changed, and what didn't?

Only median and mode remain the same 

Nobody likes Cousin Oliver. Maybe the network should have used an even younger actor. Replace Cousin Oliver with 1-year-old Jessica, then recalculate again. Does this change your choice of central tendency or variance estimation methods?

Replacing Cousin Oliver with Jessica would change mean and standard deviation.

On the 50th anniversary of The Brady Bunch, four different magazines asked their readers whether they were fans of the show. The answers were: TV Guide 20% fans Entertainment Weekly 23% fans Pop Culture Today 17% fans SciPhi Phanatic 5% fans

Based on these numbers, what percentage of adult Americans would you estimate were Brady Bunch fans on the 50th anniversary of the show?

In [31]:
fan = np.array([20, 23, 17, 5])

np.std(fan, ddof=1)/np.sqrt(len(fan))

3.944933459514875

In [35]:
np.mean(fan)

16.25

With a standard error of 3.94, I would estimate 12.31-20.19% of adult Americans were Brady Bunch fans on the 50th anniversary of the show. 